python/pyspark/worker.py

```diff
@@ -122,6 +123,10 @@ def read_single_udf(pickleSer, infile, eval_type):
         else:
             row_func = chain(row_func, f)
 
+    # make sure StopIteration's raised in the user code are not ignored
+    # when they are processed in a for loop, raise them as RuntimeError's instead
+    row_func = fail_on_stopiteration(row_func)
```

Comment:
@e-dorigatti, I think it's fine to name it `func`. Let's reduce the diff so that other backports make less conflicts in the future.